core/src/world/place/location/geographical/canyon.rs

```diff
@@ -0,0 +1,176 @@
+use crate::world::{Demographics, Place};
+use rand::prelude::*;
+
+pub fn generate(place: &mut Place, rng: &mut impl Rng, _demographics: &Demographics) {
+    place.name.replace_with(|_| name(rng));
+}
+
+fn name(rng: &mut impl Rng) -> String {
+    match rng.gen_range(0..=5) {
+        0 => format!("{} {}", thing(rng), canyon_synonym(rng)),
+        1 => format!("The {} {}", placement(rng), canyon_synonym(rng)),
+        2 => format!("{} {}", cardinal_direction(rng), canyon_synonym(rng)),
+        3 => format!("{} {}", adjective(rng), canyon_synonym(rng)),
+        4 => format!("{} {} {}", adjective(rng), thing(rng), canyon_synonym(rng)),
+        5 => format!("{}'s {}", profession(rng), canyon_synonym(rng)),
+        _ => unreachable!(),
+    }
```

Comment:
Speaking of weighting, I would say that `placement()` and `adjective()`, and maybe `cardinal_direction()`, are overrepresented in the results. Here we do have the benefit of actual weighting logic.

Reply:
I basically followed the format of the [second example](https://docs.rs/rand/0.8.5/rand/distributions/struct.WeightedIndex.html) for weighted index but I'm not sure I like how it looks. Perhaps there's a better way?

Reply:
I'm going to be AFK for a bit but I'll be back later this evening.

Reply:
I didn't have anything so fancy in mind, just:

```rust
    match rng.gen_range(0..=8) {
        0..=1 => format!("{} {}", thing(rng), canyon_synonym(rng)),
        2 => format!("The {} {}", placement(rng), canyon_synonym(rng)),
        3 => format!("{} {}", cardinal_direction(rng), canyon_synonym(rng)),
        4 => format!("{} {}", adjective(rng), canyon_synonym(rng)),
        5..=6 => format!("{} {} {}", adjective(rng), thing(rng), canyon_synonym(rng)),
        7..=8 => format!("{}'s {}", profession(rng), canyon_synonym(rng)),
        _ => unreachable!(),
    }
```

Reply:
Alright, back to the non-fancy solution. :) I went ahead and also took the liberty of demoting adjective + thing + canyon_synonym as well.

Reply:
You're the boss. Mostly I just prefer that vocabulary with fewer possibilities occurs less often, otherwise the result ends up looking quite repetitive.